### PR TITLE
Fix bulk edit and bulk delete filtering

### PR DESCRIPTION
**Pull request by snaselj**
# Closes: #4033, #4478

## What's Changed

Fixed filtering objects in bulk edit and bulk delete.

## To Do

- [x] Add bulk edit all with filter test.
    - [x] Fix skipped models.
- [x] Add bulk delete all with filter test.
    - [x] Fix skipped models.

### CODE REVIEW

**Comment by snaselj** [comment]
> Changes look good. Anything we can/should add to the generic view test cases that would have caught this bug when it was introduced?

Thanks, I can add tests covering this if desirable.

File: nautobot/core/testing/views.py [hunk 1]
```diff
@@ -1054,6 +1054,52 @@ def test_bulk_edit_form_contains_all_pks(self):
             for pk in pk_list:
                 self.assertIn(f'<input type="hidden" name="pk" value="{pk}"', response_body)
 
+        @override_settings(EXEMPT_VIEW_PERMISSIONS=["*"])
+        def test_bulk_edit_form_contains_all_filtered(self):
```
Comment (snaselj) [inline nautobot/core/testing/views.py:1058]: Added a test for bulk edit with the filter, some models fails, so skipping them so far. WDYT about that @glennmatthews pls?

File: nautobot/core/testing/views.py [hunk 2]
```diff
@@ -1054,6 +1054,52 @@ def test_bulk_edit_form_contains_all_pks(self):
...
+            # We are testing the intermediary step of bulk_edit with pagination applied and additional filter.
+            # i.e. "_all" passed in the form and filter using query params.
+            # Prequisity is not to have `bulk_edit_data` values in objects created by `setUpTestData`.
+            pk_list = self._get_queryset().values_list("pk", flat=True)
+            self.assertTrue(len(pk_list) > 1)
+            obj_perm = users_models.ObjectPermission(name="Test permission", actions=["change"])
+            obj_perm.save()
+            obj_perm.users.add(self.user)
+            obj_perm.object_types.add(ContentType.objects.get_for_model(self.model))
```
Comment (glennmatthews) [inline nautobot/core/testing/views.py:1067]: Should be able to replace these lines with something like `self.add_permissions(f"{self.model._meta.app_label}.change_{self.model._meta.model_name}")`

File: nautobot/core/testing/views.py [hunk 3]
```diff
@@ -1054,6 +1054,52 @@ def test_bulk_edit_form_contains_all_pks(self):
...
+            # Then we bulk update objects with filter applied.
+            #     We only pass in another pk to test the functionality of "_all" and filter,
+            #     which should grab instance updated in the previous step.
+            update_data = {
+                "pk": pk_list[1:2],
+                "_all": "on",
+                **post_data,
+            }
+            query_string = urlencode(post_data, doseq=True)
+            response = self.client.post(f"{self._get_url('bulk_edit')}?{query_string}", update_data)
```
Comment (glennmatthews) [inline nautobot/core/testing/views.py:1094]: Need to either:

1. restrict the constructed `query_string` to only include valid filter parameters
2. `@override_settings(STRICT_FILTERING=False)`

Probably the latter is simpler.

Comment (snaselj) [reply nautobot/core/testing/views.py]: Not necessary after switching to pk filtering.

File: nautobot/core/testing/views.py (hunk 2 above)
Comment (snaselj) [reply nautobot/core/testing/views.py]: Thanks applied.

**Comment by snaselj** [comment]
Ready for review now.

**Comment by jathanism** [comment]
> Ready for review now.

Looking good overall but I would like to see the tests passing so I can commit to final review!

**Review by jathanism — APPROVED** [review]
:shipit:

**Pull request merged**